sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -257,59 +330,153 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
 
       // lane following to intersection
-      double time_to_schedule_stop = (scheduled_stop_time_ - street_msg_timestamp_)*1000.0;
-      int case_num = determineSpeedProfileCase(stop_intersection_down_track, current_state.speed, time_to_schedule_stop, speed_limit_);
+      
+      ROS_DEBUG_STREAM("time_to_schedule_stop  " << time_to_schedule_stop);
+      int case_num = determineSpeedProfileCase(distance_to_stopline, current_state.speed, time_to_schedule_stop, speed_limit_);
+      ROS_DEBUG_STREAM("case_num:  " << case_num);
+
+      if (case_num < 3)
+      {
+        resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
+          case_num, current_downtrack_, stop_intersection_down_track-config_.stop_line_buffer, current_state.speed, 0.0,
+          current_state.stamp, time_to_schedule_stop,
+          lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
+
+      }
+      else
+      {
+        ROS_DEBUG_STREAM("Decelerating to stop at the intersection");
+
+        // at the stop line or decelerating to stop line
+        // stop and wait maneuver
+       
+        std::vector<lanelet::ConstLanelet> crossed_lanelets =
+            getLaneletsBetweenWithException(current_downtrack_, stop_intersection_down_track, true, true);
+        
+
+        double stopping_accel = caseThreeSpeedProfile(stop_intersection_down_track-current_state.downtrack, current_state.speed, time_to_schedule_stop);
+        // TODO: temp val for acc- remove when carma streets is running
+        stopping_accel = 1.5;
+        resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+          current_state.downtrack, stop_intersection_down_track-config_.stop_line_buffer, current_state.speed, crossed_lanelets[0].id(),
+          crossed_lanelets[0].id(), stopping_accel, current_state.stamp,
+          current_state.stamp + ros::Duration(time_to_schedule_stop)));
+
+        
+      }
+  }
 
-      resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
-        case_num, current_downtrack_, stop_intersection_down_track, current_state.speed, 0.0,
-        current_state.stamp, time_to_schedule_stop,
-        lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
-    }
-    else
-    {
-      // at the stop line
-      // stop and wait maneuver
-      auto stop_line_lanelet = nearest_stop_intersection->lanelets().front();
-      double stop_duration = (scheduled_enter_time_ - scheduled_stop_time_)*1000.0;
-      ROS_DEBUG_STREAM("Planning stop and wait maneuver");
-      resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
-        current_downtrack_, stop_intersection_down_track, current_state.speed, stop_line_lanelet.id(),
-        stop_line_lanelet.id(), current_state.stamp,
-        current_state.stamp + ros::Duration(stop_duration)));
+  bool cond1 = (int(street_msg_timestamp_) - int(scheduled_stop_time_)) > 0;
```

Comment:
Is there a reason for this int casts? I'm not sure what role they serve, if we're looking for some kind of an epsilon factor I'd rather be explicit about it `if (street_msg_timestamp - scheduled_stop_time > EPSLION) { ...` for clarity's sake than relying on datatype conversion to do it.

Also cond1 and cond2 could be named better, to make it more clear what's being checked and why it relates to having enough time to stop. If not renaming them, at least a comment would be helpful to describe what this block of logic does.

Reply:
As mentioned above, the regular difference is not giving the result I expect. But rename the conditions to be more clear. Basically, we are using the schedule timestamps to know what to do at the intersection, and when to do it.